packages/types/src/events/message.ts

```diff
@@ -0,0 +1,368 @@
+}
+
+export interface ChannelUnarchiveMessageEvent {
+  type: 'message';
+  subtype: 'channel_unarchive';
+  team: string;
+  user: string;
+  channel: string;
+  channel_type: ChannelTypes;
+  text: string;
+  ts: string;
+  event_ts: string;
+}
+
+export interface EKMAccessDeniedMessageEvent {
+  type: 'message';
+  subtype: 'ekm_access_denied';
+  event_ts: string;
+  channel: string;
+  channel_type: ChannelTypes;
+  ts: string;
+  text: string; // This will not have any meaningful content within
+  user: 'UREVOKEDU';
+}
+
+export interface FileShareMessageEvent {
+  type: 'message';
+  subtype: 'file_share';
+  text: string;
+  attachments?: MessageAttachment[];
+  blocks?: (KnownBlock | Block)[];
+  files?: File[];
+  upload?: boolean;
+  display_as_bot?: boolean;
+  x_files?: string[];
+  user: string;
+  parent_user_id?: string;
+  ts: string;
+  thread_ts?: string;
+  channel: string;
+  channel_type: ChannelTypes;
+  event_ts: string;
+}
+
+export interface MeMessageEvent {
+  type: 'message';
+  subtype: 'me_message';
+  event_ts: string;
+  channel: string;
+  channel_type: ChannelTypes;
+  user: string;
+  text: string;
+  ts: string;
+}
+
+export interface MessageChangedEvent {
+  type: 'message';
+  subtype: 'message_changed';
+  event_ts: string;
+  hidden: true;
+  channel: string;
+  channel_type: ChannelTypes;
+  ts: string;
+  message: MessageEvent;
+  previous_message: MessageEvent;
+}
+
+export interface MessageDeletedEvent {
+  type: 'message';
+  subtype: 'message_deleted';
+  event_ts: string;
+  hidden: true;
+  channel: string;
+  channel_type: ChannelTypes;
+  ts: string;
+  deleted_ts: string;
+  previous_message: MessageEvent;
+}
+
+export interface MessageRepliedEvent {
+  type: 'message';
+  subtype: 'message_replied';
+  event_ts: string;
+  hidden: true;
+  channel: string;
+  channel_type: ChannelTypes;
+  ts: string;
+  message: MessageEvent & {
+    // TODO: should this be the union of all message events with type 'message'?
+    thread_ts: string;
+    reply_count: number;
+    replies: MessageEvent[]; // TODO: should this be the union of all message events with type 'message'?
+  };
+}
+
+// the `reply_broadcast` message subtype is omitted because it is discontinued
+
+export interface ThreadBroadcastMessageEvent {
+  type: 'message';
+  subtype: 'thread_broadcast';
+  event_ts: string;
+  text: string;
+  attachments?: MessageAttachment[];
+  blocks?: (KnownBlock | Block)[];
+  user: string;
+  ts: string;
+  thread_ts?: string;
+  root: (GenericMessageEvent | BotMessageEvent) & {
+    thread_ts: string;
+    reply_count: number;
+    reply_users_count: number;
+    latest_reply: string;
+    reply_users: string[];
+  };
+  client_msg_id: string;
+  channel: string;
+  channel_type: ChannelTypes;
+}
+
+interface BotProfile {
+  id: string;
+  name: string;
+  app_id: string;
+  team_id: string;
+  icons: {
+    [size: string]: string;
+  };
+  updated: number;
+  deleted: boolean;
+}
+
+interface File {
```

Comment:
Should this be in `file.ts` instead?

Reply:
No, because `file.ts` holds the file-related event shapes, while here within `message.ts` we house message event related shapes, which look like contain file-related information (thus the existence of this interface).

I think once I get to the refactoring task (https://github.com/slackapi/node-slack-sdk/issues/1904), I will take a look at the entire types package holistically and see where else we need a `File` interface and make sure it's reused appropriately.